gendo/bot.py

```diff
@@ -175,6 +196,31 @@ def add_listener(self, rule, view_func=None, func_args=None,
             raise TypeError('view_func should be callable')
         self.listeners.append((rule, view_func, func_args, decorator_options))
 
+    def add_handler(self, event_name, f=None, args=None, decorator_options=None):
+        # FIXME: What?
+        handlers = self.handlers.setdefault(event_name, [])
+        handlers.append((f, args, decorator_options))
+
+    def handle(self, event_name, data):
+        handlers = self.handlers.get(event_name, [])
+        for function, args, decorator_options in handlers:
+            user, channel, response = function(data, **args)
+            if not response:
```

Comment:
This block is the same as `respond`. Could be pulled out into a separate method.